lib/rules/no-global-setup.js

```diff
@@ -0,0 +1,27 @@
+'use strict'
+
+/**
+ * @fileoverview Disallow using setup and teardown methods outside a suite
+ * @author Omer Ganim
+ */
+
+var suiteRegexp = /^(f|d|x)?describe$/
+var setupRegexp = /^(before|after)(Each|All)$/
+
+module.exports = function (context) {
+  var suiteDepth = 0
+  return {
+    CallExpression: function (node) {
+      if (suiteRegexp.test(node.callee.name)) {
+        suiteDepth++
```

Comment:
In my experience in [eslint-plugin-lodash](https://github.com/wix/eslint-plugin-lodash), it's significantly more performant checking a variable every visit instead of going up the tree.
Also, the visit order is part of the spec, not an implementation detail, and ESLint relies on it as well. (Here are [two](https://github.com/eslint/eslint/blob/master/lib/rules/array-callback-return.js#L214) [examples](https://github.com/eslint/eslint/blob/master/lib/rules/max-depth.js))